pkg/vsphere/simulator/property_collector.go

```diff
@@ -40,6 +41,37 @@ func NewPropertyCollector(ref types.ManagedObjectReference) object.Reference {
 var errMissingField = errors.New("missing field")
 var errEmptyField = errors.New("empty field")
 
+func getObject(ref types.ManagedObjectReference) (reflect.Value, bool) {
+	obj := Map.Get(ref)
+	if obj == nil {
+		return reflect.Value{}, false
+	}
+
+	rval := reflect.ValueOf(obj).Elem()
+	rtype := rval.Type()
+
+	// PropertyCollector is for Managed Object types only (package mo).
```

Comment:
We had already been behaving the same way as vSphere in that respect, by virtue of how Go's reflection works.